circuit_definitions/README.md

```diff
@@ -0,0 +1,55 @@
+# Circuits definitions crate
+
+This crate contains the 'outer layer' for multiple circuits. The concrete circuits code is in `era-zkevm_circuits`
+repository.
+
+## Code structure
+
+We have 13 different 'base layer' circuits (for example MainVM, Decomitter), and 3 recursive circuits (Leaf, Node and
+Scheduler).
+
+Base layer circuits are located in `src/base_layer`, Recursive circuits are in `src/recursion_layer`.
+
+We also have 2 'AUX' circuits: compressor and wrapper, that are run on top of the final Scheduler proof, and they are
+located in `src/aux_layer`.
+
+![circuits](https://user-images.githubusercontent.com/128217157/275817097-0a543476-52e5-437b-a7d3-10603d5833fa.png)
+
+`src/encodings` directory contain some helper structs that are used by the test harness (and should match the ones used
+in circuits themselves).
+
+## Circuit types
+
+We have 12 different circuit types (in witness, you might notice 13, as one circuit (events_dedup_and_sort) is used for
+both L1 messages and events).
+
+| Circuit name             | Location                     | Description                                                                                                                                                          |
+| ------------------------ | ---------------------------- | -------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| Main VM                  | vm_main.rs                   | Executes OpCodes from the VM                                                                                                                                         |
+| CodeDecommittmentsSorter | sort_code_decommit.rs        | Verifies the order of requests for code decommitment (fetching bytecode based on hash).                                                                              |
+| CodeDecommiter           | code_decommiter.rs           | Verifies actual code decommitment - that the bytecode stored in a given memory location, has a correct hash                                                          |
```

Comment:
As far as I know, it doesn't check that the bytecode is located in memory, but write it instead